### ISSUE TITLE
Probable case not considered

### ISSUE THREAD

**Issue by obhalerao97** [ankan-ban/llama2.cu#6]
Hello, 
Have you checked for what happens when the n_heads != n_kv_heads? How does this affect the Rope rotation, MHA which now becomes GQA?

**Comment by ankan-ban** [comment]
Yes, the code should work when n_heads != n_kv_heads. I have tested it with codellama-34b model that uses GQA.

**Comment by ankan-ban** [comment]
Sorry, please ignore the previous comment. I mistook this with a different repo. I am no longer maintaining this repo. All new development is happening at https://github.com/ankan-ban/llama_cu_awq which supports GQA.

**Comment by obhalerao97** [comment]
Okay thank you! Is there a version of the code that's able to process a batch of strings?
